### DISCUSSION TITLE
Why when I change routing option the itinerary stay the same ?

### DISCUSSION THREAD

**Discussion by jclaudej** [devemux86/cruiser#83] [Q&A] [answered]
Hi,

With Kurviger pro when I change routing option the itinerary is automaticly adapt. For exemple with or with motorway.
With Cruiser I must to cancel my itinerary and after select the good option for my travel and afetr I can define my itinerary. 
Ithink that the Kurviger pro solution is more comfortable.
Can I hope an update with this functionnality ?
An other point that isn't clear for my small and old head
Whane I select :
Parameter/routing/address search I get for example :
GraphHopper
OpenCage
If I understand this function is for find an adress and I would like to know how choice this variable ?

By adavance, thank a lot for your help

Jean-Claude

**Comment by devemux86** [comment]
Hi @jclaudej 

Routing options do not automatically calculate new routes in Cruiser. They are used for the next route calculations.
You can trigger a new route calculation easily: long press on any waypoint, select Move, press OK without moving.

However, I am open to changes if users need different functionality?

---

The search provider can be selected in "Settings | Routing | Search" or search settings for "search".
It affects how you can search online for locations, each provider works differently, better test them?

---

Cruiser search is currently being rewritten to add **offline search**, favorites search. Please see here:
- https://github.com/devemux86/cruiser/discussions/74

**Reply by jclaudej** [reply]
I think that a same routing option than Kurviger pro will be a positive point because you can image that when you create an itinerary sometime you want quickly know the different time of road in function of your routing options.
Perhaps other people could give thier point of view about it.

Thank a lot for your reactivity ;)

**Reply by 0709wiwiwi** [reply]
Kurviger Pro MAIN routing is by the kurviger online engine. 
In the Cruiser app multiple routing engines are offered. 
In Cruiser, you are thus more likely to make "wrong" choices.
Accidental damage by a recalculation should be prevented.
An undo redo button could fastly repair accidental damage.
An undo redo button is a also nice to fastly compare results.
Cruiser offers a menu history. An undo redo is more simple.

_Extra buttons yes but darned handy if you were momentarily confused by those many choices._

**Reply by devemux86** [reply]
> Cruiser offers a menu history. An undo redo is more simple.

This is already in my implementation list. :slightly_smiling_face:

**Comment by devemux86** [comment]
A safe solution, avoiding automatic calculation when change router or options, is to add a "Recalculate" option in route menu:

(these menus could be changed with the new icon screens)

<img src="https://github.com/devemux86/cruiser/assets/3484020/5528d288-fe79-4322-8955-c8a79187770a" width="33%"/>

**Comment by devemux86** [comment, marked as answer]
"Recalculate route" action is available in route menu in [Cruiser 3.0.22](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
